components/navigation/EventPostItem.tsx

```diff
@@ -0,0 +1,99 @@
+import { ArrowRightIcon } from '@heroicons/react/outline';
+import moment from 'moment';
+import React from 'react';
+
+import { HeadingLevel, HeadingTypeStyle } from '@/types/typography/Heading';
+
+import IconCalendar from '../icons/Calendar';
+import Community from '../icons/Community';
+import Conference from '../icons/Conference';
+import Webinar from '../icons/Webinar';
+import Heading from '../typography/Heading';
+
+interface Event {
+  title: string;
+  url: string;
+  banner?: string;
+  date: string;
+}
+
+interface EventPostItemProps {
+  post: Event;
+  className?: string;
+  id: string;
+}
+
+/**
+ * @description Component representing an event post item.
+ * @param {EventPostItemProps} props - The props for the EventPostItem component.
+ */
+function EventPostItem({ post, className = '', id }: EventPostItemProps): JSX.Element {
+  const localTime = moment().format('YYYY-MM-DD'); // store localTime
+  const currentDate = `${localTime  }T00:00:00.000Z`;
+  const title = post.title || '';
+  let color = '';
+  let icon = <></>;
```

Comment:
Instead of defining it as empty JSX, make a `IconType` and use it as a type here.